external-crates/move/crates/move-compiler/src/parser/lexer.rs

```diff
@@ -463,63 +497,42 @@ impl<'input> Lexer<'input> {
     // Calling this function during parsing effectively marks a valid point for documentation
     // comments. The documentation comments are not stored in the AST, but can be retrieved by
     // using the start position of an item as an index into `matched_doc_comments`.
-    pub fn match_doc_comments(&mut self) {
-        if let Some(comments) = self.read_doc_comments() {
-            self.attach_doc_comments(comments);
-        };
+    pub fn take_doc_comment(&mut self) -> Option<(u32, u32, String)> {
+        self.current_doc_comment.take()
     }
 
-    // Matches the doc comments after the last token (or the beginning of the file) to the position
-    // of the current token. This moves the comments out of `doc_comments` and
-    // into `matched_doc_comments`. At the end of parsing, if `doc_comments` is not empty, errors
-    // for stale doc comments will be produced.
-    pub fn read_doc_comments(&mut self) -> Option<String> {
-        let start = self.previous_end_loc() as u32;
-        let end = self.cur_start as u32;
-        let mut matched = vec![];
-        let merged = self
-            .doc_comments
-            .range((start, start)..(end, end))
-            .map(|(span, s)| {
-                matched.push(*span);
-                s.clone()
-            })
-            .collect::<Vec<String>>()
-            .join("\n");
-        if !matched.is_empty() {
-            for span in matched {
-                self.doc_comments.remove(&span);
+    // Restores the doc comment that was temporarily taken by `take_doc_comment`. This is used to
+    // allow for tokens to intersperse a doc comment, like in the case of attributes `#[...]` where
+    // a doc comment can continue with an attribute in the middle, e.g.
+    // ```
+    // /// This is a doc comment for 'fun foo'
+    // #[attr]
+    // /// This is a part of the same doc comment for 'fun foo'
```

Comment:
Out of curiosity, does this work in Rust? Either way, should we lint against it?

Reply:
This works in Rust. Not sure if it is worth linting/warning. It would be a bit tricky to do

Reply:
Also important to note that this is how this worked before this PR (cc @tzakian)